larpandoracontent/LArVertex/TrainedVertexSelectionAlgorithm.cc

```diff
@@ -561,6 +640,30 @@ const pandora::Vertex *TrainedVertexSelectionAlgorithm::ProduceTrainingExamples(
         VertexFeatureInfo vertexFeatureInfo(vertexFeatureInfoMap.at(pVertex));
         this->AddVertexFeaturesToVector(vertexFeatureInfo, featureList, useRPhi);
 
+        if (m_legacyVariables)
+        {
+            LArMvaHelper::MvaFeatureVector sharedFeatureList;
+            float separation(0.f), axisHits(0.f);
+            this->GetSharedFeatures(pVertex, pBestVertex, kdTreeMap, separation, axisHits);
+            VertexSharedFeatureInfo sharedFeatureInfo(separation, axisHits);
+            this->AddSharedFeaturesToVector(sharedFeatureInfo, sharedFeatureList);
+
+            if (pBestVertex && (bestVertexDr < maxRadius))
+            {
+                if (coinFlip(generator))
+                {
+                    LArMvaHelper::ProduceTrainingExample(trainingOutputFile + "_" + interactionType + ".txt", true, eventFeatureList,
+                        bestVertexFeatureList, featureList, sharedFeatureList);
+                }
+
+                else
+                {
+                    LArMvaHelper::ProduceTrainingExample(trainingOutputFile + "_" + interactionType + ".txt", false, eventFeatureList,
+                        featureList, bestVertexFeatureList, sharedFeatureList);
+                }
+            }
+        }
+
         if (pBestVertex && (bestVertexDr < maxRadius))
```

Comment:
I'm a little confused by the second block of training example production. This seems to imply that two lots of training examples will be produced for the legacy code. Is this intended?

Reply:
Yep this is a mistake. The second set should be in an else block. It should also be `!m_legacyVariables`. This didn't scream at me locally because unlike with the operational legacy mode it will still run fine. Thanks so much for picking this one up!

Reply:
I have done some checks locally to ensure the files are producing the number of features I'm expecting. Everything looks okay now :)